### ISSUE TITLE
ImportStateFunc delivers obsolete value

### ISSUE THREAD

**Issue by abergmeier** [hashicorp/terraform-plugin-sdk#566]
### SDK version

```
v2.0.1
```

### Relevant provider source code

```go
// See https://github.com/abergmeier/terraform-provider-exasol/blob/620cc7611369e25aaf4424cf4e0f9144bd2a12ee/internal/resources/connection_test/acc_test.go

...

	resource.Test(t, resource.TestCase{
		Providers: test.DefaultAccProviders,
		Steps: []resource.TestStep{
			{
				PreConfig: tryDeleteConnection,
				Config: fmt.Sprintf(`%s
				resource "exasol_connection" "test" {
					name = "%s"
					to = "foo"
				}
				`, test.HCLProviderFromConf(&exaConf), dbName),
			},
			{
				PreConfig:         createConnection,
				ResourceName:      "exasol_connection.test",
				ImportState:       true,
				ImportStateId:     strings.ToUpper(dbName),
				ImportStateVerify: true,
				ImportStateCheck:  checkImport(dbName),
			},
		},
	})

...

func checkImport(tableName string) resource.ImportStateCheckFunc {
	return func(s []*terraform.InstanceState) error {
		if len(s) == 0 {
			return errors.New("No Instance found")
		}

		if len(s) != 1 {
			return fmt.Errorf("Expected one Instance: %d", len(s))
		}

		name := s[0].Attributes["name"]

		if name != tableName {
			return fmt.Errorf("Expected name %s: %s", tableName, name)
		}

		to := s[0].Attributes["to"]

		if to != "ftp://192.168.1.1/" {
			return fmt.Errorf("Expected to ftp://192.168.1.1/: %s", to)
		}

		return nil
	}
}

```

### Debug Output
Added Debug Output to check values and the new values are `Set`.

### Expected Behavior
`ImportStateFunc` should be provided the correct state of ResourceData

### Actual Behavior
There are still old values being delivered. Thus `checkImport` fails.

The documentation (especially implicit behavior) of the Import testing does not seem to be documented anywhere :(

**Comment by bflad** [comment]
I believe this may be already fixed per https://github.com/hashicorp/terraform-plugin-sdk/issues/531 / https://github.com/hashicorp/terraform-plugin-sdk/pull/552 and pending an SDK release.

**Comment by abergmeier** [comment]
Ok, any plans for a release?

**Comment by paddycarver** [comment]
We're currently working to resolve some issues with some merged code, and are looking to cut a release as soon as feasible after that. So: "shortly", though the timeline is dictated by the known unknown of how long it takes to resolve the bugs.

**Comment by paddycarver** [comment]
This should be resolved with the release of v2.0.2. If you continue to experience issues, please let us know, and we can reopen this issue and investigate further. Thanks!
